### ISSUE TITLE
"more than 1 Code Generation option is specified.." error 

### ISSUE THREAD

**Issue by flugenheimer** [NVlabs/instant-ngp#265]
I get this after running "cmake --build build --config RelWithDebInfo -j 16"

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\BuildCustomizations\CUDA 11.6.t
argets(630,9): error : More than 1 Code Generation option is specified, this is only allowed if NVCC Compilation Type i
s 'compile' or 'fatbin'.

**Comment by mmalex** [comment]
hi! I've seen this on one occasion, and I never got to the bottom of it. However, I found that if you switch generator to Ninja, the problem went away (for me). I'm afraid that's not a full solution, but it might be the beginning of a workaround...

**Comment by flugenheimer** [comment]
> 

Hi mmalex. I am guessing that this is what you suggest? 
https://www.scivision.dev/cmake-generator-ninja/

**Comment by Tom94** [comment]
Yep. You should be able to simply add `-GNinja` to the cmake command.

**Comment by flugenheimer** [comment]
I have tried to install ninja using pip: 
python.exe -m pip install ninja

and then in the Developer Command Prompt run: 
cmake . -B build -GNinja

This however gives me the errror: 
CUDA_ARCHITECTURES is empty for target "cmTC_2cc24".

I do not get this error without the "-GNinja"

**Comment by Tom94** [comment]
You you let us know which CMake version you have installed?

**Comment by flugenheimer** [comment]
cmake version 3.22.2
